src/nrniv/netpar.cpp

```diff
@@ -1095,15 +1100,17 @@ void BBS::outputcell(int gid) {
 void BBS::spike_record(int gid, IvocVect* spikevec, IvocVect* gidvec) {
 	PreSyn* ps;
     if (gid >= 0) {
-	nrn_assert(gid2out_->find(gid, ps));
-	assert(ps);
-	ps->record(spikevec, gidvec, gid);
+        nrn_assert(gid2out_->find(gid, ps));
+        assert(ps);
+        ps->record(spikevec, gidvec, gid);
     }else{ // record all output spikes
-	NrnHashIterate(Gid2PreSyn, gid2out_, PreSyn*, ps) {
-		if (ps->output_index_ >= 0) {
-			ps->record(spikevec, gidvec, ps->output_index_);
-		}
-	}}}
+        all_spiketvec = spikevec;
```

Comment:
We should consider reference counting all_spiketvec and all_spikegidvec through their obj_  field (if obj_ is non NULL).
In Vector.record and Vector.play this is accomplished through a more complicated InterViews Observer/Observable protocol but I don't recommend using that.

Reply:
I believe Michael is referring to the usage of underlying vector, when we come back from CORENEURON make sure that this vector is still used (refcount > 0).